iOS_SDK/OneSignalSDK/Source/Requests.m

```diff
@@ -201,36 +201,45 @@ + (instancetype)withUserId:(NSString *)userId appId:(NSString *)appId deviceToke
 @end
 
 @implementation OSRequestCreateDevice
-+ (instancetype _Nonnull)withAppId:(NSString * _Nonnull)appId withDeviceType:(NSNumber * _Nonnull)deviceType withEmail:(NSString * _Nullable)email withPlayerId:(NSString * _Nullable)playerId withEmailAuthHash:(NSString * _Nullable)emailAuthHash withExternalIdAuthToken:(NSString * _Nullable)externalIdAuthToken {
++ (instancetype _Nonnull)withAppId:(NSString * _Nonnull)appId withDeviceType:(NSNumber * _Nonnull)deviceType withEmail:(NSString * _Nullable)email withPlayerId:(NSString * _Nullable)playerId withEmailAuthHash:(NSString * _Nullable)emailAuthHash withExternalUserId: (NSString * _Nullable)externalUserId withExternalIdAuthToken:(NSString * _Nullable)externalIdAuthToken {
     let request = [OSRequestCreateDevice new];
     
-    request.parameters = @{
-       @"app_id" : appId,
-       @"device_type" : deviceType,
-       @"identifier" : email ?: [NSNull null],
-       @"email_auth_hash" : emailAuthHash ?: [NSNull null],
-       @"external_user_id_auth_hash" : externalIdAuthToken ?: [NSNull null],
-       @"device_player_id" : playerId ?: [NSNull null]
-    };
-    
+    let params = [[NSMutableDictionary alloc] initWithDictionary:@{
+        @"app_id" : appId,
+        @"device_type" : deviceType,
+        @"identifier" : email ?: [NSNull null],
+        @"email_auth_hash" : emailAuthHash ?: [NSNull null],
+        @"external_user_id_auth_hash" : externalIdAuthToken ?: [NSNull null],
+        @"device_player_id" : playerId ?: [NSNull null]
+     }];
+    
+    if (externalUserId) {
+        params[@"external_user_id"] = externalUserId;
```

Comment:
Curious why you needed to deviate from the pattern for setting params (L208-213). I'm primarily interested because I'm not very familiar with this language and am trying to learn.

Is it something to do with not including that parameter in the body if there's no value?

```json
{
    "app_id": "123",
    "external_user_id": "" //null
}
```

###### vs

```json
{
   "app_id": "123",
}
```
How were the request params included in the past? I didn't see a statement setting params in the past commit. Is there another way of setting other than `request.parameters = params;` ?